services/self-service/src/main/java/com/epam/datalab/backendapi/service/impl/BillingServiceImpl.java

```diff
@@ -340,6 +341,21 @@ private void appendStatuses(BillingReportLine br) {
         }
     }
 
+    private void appendShapes(BillingReportLine br) {
+        BillingResourceType resourceType = br.getResourceType();
+        if (BillingResourceType.COMPUTATIONAL == resourceType) {
+            String shape = "Master: 1 x %s Slave: %s x %s";
+            exploratoryService.getUserInstance(br.getUser(), br.getProject(), br.getExploratoryName(), true)
+                    .flatMap(ui -> ui.getResources()
+                            .stream()
+                            .filter(cr -> cr.getComputationalName().equals(br.getResourceName()))
+                            .findAny())
+                    .ifPresent(cr -> br.setShape(
+                            String.format(shape, cr.getMasterNodeShape(), cr.getTotalInstanceCount() - 1, cr.getSlaveNodeShape())
```

Comment:
May I clarify, why the instances count is decreased by 1?

Reply:
Total instance count contains sum of number of slave nodes and master nodes. So to get number of slave nodes we need to subtract number of master nodes. Added java doc and variable for readability. Thanks for suggestion.